hydrolib/core/dflowfm/extold/models.py

```diff
@@ -0,0 +1,417 @@
+    3. Interpolate time and space, save weights
+    4. Interpolate space
+    5. Interpolate time
+    7. Interpolate/Extrapolate time
+    """
+
+    operand: Operand = Field(alias="OPERAND")
+    """Operand: Overwriting or superimposing values already set for this quantity:
+    'O' Values are overwritten.
+    '+' New value is superimposed.
+    """
+
+    value: Optional[float] = Field(None, alias="VALUE")
+    """Optional[float]: custom coefficients for transformation."""
+
+    factor: Optional[float] = Field(None, alias="FACTOR")
+    """Optional[float]: The factor."""
+
+    ifrctyp: Optional[float] = Field(None, alias="IFRCTYP")
+    """Optional[float]: The friction type."""
+
+    averagingtype: Optional[float] = Field(None, alias="AVERAGINGTYPE")
+    """Optional[float]: The averging type."""
+
+    relativesearchcellsize: Optional[float] = Field(
+        None, alias="RELATIVESEARCHCELLSIZE"
+    )
+    """Optional[float]: The relative search cell size."""
+
+    extrapoltol: Optional[float] = Field(None, alias="EXTRAPOLTOL")
+    """Optional[float]: The extrapolation tolerance."""
+
+    percentileminmax: Optional[float] = Field(None, alias="PERCENTILEMINMAX")
+    """Optional[float]: The percentile min max."""
+
+    area: Optional[float] = Field(None, alias="AREA")
+    """Optional[float]: The area for sources and sinks."""
+
+    nummin: Optional[int] = Field(None, alias="NUMMIN")
+    """Optional[int]: The nummin."""
+
+    @validator("quantity", pre=True)
+    def validate_quantity(cls, value):
+        if isinstance(value, str):
+            lower_value = value.lower()
+
+            if lower_value.startswith(Quantity.TracerBnd):
+                n = len(Quantity.TracerBnd.value)
+                return Quantity.TracerBnd.value + value[n:]
+
+            if lower_value.startswith(Quantity.InitialTracer):
+                n = len(Quantity.InitialTracer.value)
+                return Quantity.InitialTracer.value + value[n:]
+
+            supported_values = list(Quantity)
+            if lower_value in supported_values:
+                return Quantity(lower_value)
+
+            supported_value_str = ", ".join(([x.value for x in supported_values]))
+            raise ValueError(
+                f"QUANTITY '{value}' not supported. Supported values: {supported_value_str}"
+            )
+
+        return value
+
+    @validator("operand", pre=True)
+    def validate_operand(cls, value):
+        if isinstance(value, str):
+
+            for operand in Operand:
+                if value.lower() == operand.value.lower():
+                    return operand
+
+            supported_value_str = ", ".join(([x.value for x in Operand]))
+            raise ValueError(
+                f"OPERAND '{value}' not supported. Supported values: {supported_value_str}"
+            )
+
+        return value
+
+    @root_validator(skip_on_failure=True)
+    def validate_forcing(cls, values):
+        def alias(field_key: str):
```

Comment:
Nice, clean ✨

Reply:
Thanks, it became a helluva function though....